Neo4j.Driver/Neo4j.Driver/Internal/Metrics/IConnectionListener.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) 2002-2018 "Neo Technology,"
+// Network Engine for Objects in Lund AB [http://neotechnology.com]
+//
+// This file is part of Neo4j.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Diagnostics;
+
+namespace Neo4j.Driver.Internal.Metrics
+{
+    internal interface IConnectionListener
```

Comment:
Maybe add `BeforeDisconnect` and `AfterDisconnect` to measure hard disconnects, does it makes sense?

Reply:
Feels not so useful to have the timer on Connection.conn actually. So I would hold to do this in the first version and then leave it in next iteration if you also agree?

Reply:
Yeah, ok. The amount of time spent closing the connection really doesn't make sense. I somehow thought about the number of connections hard closed. Never mind, we can leave it for next iteration as you suggested.